src/go/middleware/middleware.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/magma/magma/src/go/capture/buffer"
+	"github.com/magma/magma/src/go/protos/magma/capture"
+	"github.com/magma/magma/src/go/protos/magma/config"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+//go:generate go run github.com/golang/mock/mockgen -source=middleware.go -package mock_middleware -destination mock_middleware/mock_middleware_test.go Middleware
+
+// middleware defines middlewares functions - used for generating mocks.
+type middleware interface {
+	GetDialOptions() []grpc.DialOption
+	GetServerOptions() []grpc.ServerOption
+	withCaptureUnaryClientInterceptor() grpc.DialOption
+	withCaptureUnaryServerInterceptor() grpc.ServerOption
+}
+
+type Middleware struct {
```

Comment:
did you mean to define the interface publicly and leave the impl private?

Reply:
flipped it